### PR TITLE
Add local_addr method to TcpBuilder #26

### PR DESCRIPTION
**Pull request by nbaksalyar**
This PR implements a feature requested in #26.

I'm not quite sure about the Windows part of the implementation but I haven't found a better way to do that. Besides, AFAIK, in [libstd it's implemented](https://github.com/rust-lang/rust/blob/1.15.1/src/libstd/sys/windows/c.rs#L783) in a similar way.

### CODE REVIEW

File: src/sys/windows/mod.rs [hunk 1]
```diff
@@ -24,10 +24,42 @@ const WSA_FLAG_OVERLAPPED: DWORD = 0x01;
...
+
+    #[repr(C)]
+    #[derive(Copy, Clone)]
+    pub struct sockaddr_in6 {
+        pub sin6_family: ADDRESS_FAMILY,
+        pub sin6_port: USHORT,
+        pub sin6_flowinfo: c_ulong,
+        pub sin6_addr: in6_addr,
+        pub sin6_scope_id: c_ulong,
+    }
```
Comment (alexcrichton) [inline src/sys/windows/mod.rs]: Could these be pulled in from winapi instead?

**Comment by alexcrichton** [comment]
Looks great, thanks! Just one minor nit but otherwise r=me

File: src/sys/windows/mod.rs (hunk 1 above)
Comment (nbaksalyar) [reply src/sys/windows/mod.rs]: @alexcrichton, unfortunately there's no `sockaddr_in6` defined in `winapi`.
However there is `sockaddr_in`. Should I send a PR to `winapi` first?

Comment (alexcrichton) [reply src/sys/windows/mod.rs]: Yeah if possible let's try to get these upstream first

Comment (nbaksalyar) [reply src/sys/windows/mod.rs]: Whoops, sorry - I looked at the wrong branch, `dev`, which is work-in-progress as of now. winapi v.0.2 _does_ have the required structures that can be imported. Thanks a lot for your help Alex! :)

**Pull request merged**